### PR TITLE
Root out SCXML-related wrappers

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

**Review by davidkpiano — COMMENTED** [review]
Just skimmed this; it's looking good 👍

File: packages/core/src/StateMachine.ts [hunk 1]
```diff
@@ -237,23 +233,20 @@ export class StateMachine<
...
     if (
-      isSCXMLErrorEvent(scxmlEvent) &&
-      !currentState.nextEvents.some(
-        (nextEvent) => nextEvent === scxmlEvent.name
-      )
+      isErrorEvent(event) &&
+      !currentState.nextEvents.some((nextEvent) => nextEvent === event.type)
     ) {
-      throw scxmlEvent.data.data;
+      throw (event as any).data;
```
Comment (davidkpiano) [inline packages/core/src/StateMachine.ts:246]:
```suggestion
      // TODO: type as an error event
      throw (event as any).data;
```

**Review by davidkpiano — APPROVED** [review]
Some nits, LGTM 🚢

File: packages/core/src/actors/promise.ts [hunk 2]
```diff
@@ -30,27 +29,23 @@ export function fromPromise<T>(
...
-            data: eventObject.data,
+            data: (event as any).data,
             input: undefined
           };
         case rejectEventType:
           return {
             ...state,
             status: 'error',
-            data: eventObject.data,
+            data: (event as any).data,
```
[the remark this answers is not in the document]
Comment (Andarist) [reply packages/core/src/actors/promise.ts]: managed to fix this in a different way now, thanks for the call out

**Pull request merged**